File: Compiler/Code/bench/multi_lang_benchmark/w2_sieve/Sieve.java
public class Sieve {
    public static void main(String[] args) {
        long t0 = System.nanoTime();
        int n = 100000;
        int[] flags = new int[n + 1];

        for (int p = 2; p < 320; ++p) {
            if (flags[p] == 0) {
                int mult = p * p;
                while (mult <= n) {
                    flags[mult] = 1;
                    mult += p;
                }
            }
        }

        int count = 0;
        for (int i = 2; i <= n; ++i) {
            if (flags[i] == 0) {
                count++;
            }
        }

        long t1 = System.nanoTime();
        long us = (t1 - t0) / 1000;
        System.out.println("W2_SIEVE checksum=" + count + " time_us=" + us);
    }
}

File: Compiler/Code/bench/multi_lang_benchmark/w2_sieve/sieve.py
import time

def main():
    t0 = time.perf_counter()
    n = 100000
    flags = [0] * (n + 1)

    for p in range(2, 320):
        if flags[p] == 0:
            mult = p * p
            while mult <= n:
                flags[mult] = 1
                mult += p

    count = 0
    for i in range(2, n + 1):
        if flags[i] == 0:
            count += 1

    t1 = time.perf_counter()
    us = int((t1 - t0) * 1_000_000)
    print(f"W2_SIEVE checksum={count} time_us={us}")

if __name__ == "__main__":
    main()
